app/services/course/user_registration_service.rb

```diff
@@ -15,34 +15,86 @@ def register(registration)
 
   private
 
-  # Registers the given +user+ for a +course+. This sets the course user to the +requested+ state.
+  # Registers the given +user+ for a +course+. This sets the course user to the +requested+
+  # state, unless an explicit +workflow_state+ is passed in the +options+.
+  #
+  # This also sets the given +user+'s state as specified in the +options+ if the user already
+  # exists in the database.
   #
   # @param [Course::Registration] registration The registration model containing the course user
   #   parameters.
+  # @param [Hash] options Additional options for creating the course user.
   # @return [bool] True if the creation succeeded.
-  def register_course_user(registration)
-    course_user = CourseUser.new(course: registration.course, user: registration.user,
-                                 creator: registration.user, updater: registration.user)
-    registration.course_user = course_user
-    course_user.save
+  def register_course_user(registration, options = {}.freeze)
```

Comment:
Assignment Branch Condition size for register_course_user is too high. [17.15/15]